whatshap/cli/stats.py

```diff
@@ -341,13 +345,17 @@ def parse_variant_tables(vcf_reader, chromosomes=None):
         yield from vcf_reader
 
 
-def get_chr_lengths(chr_lengths_file, vcf_reader):
+def get_chr_lengths(vcf_reader: VcfReader, chr_lengths_file: Optional[str] = None) -> Dict[str, int]:
+    """
+    Try to map chromosome length to its name. Use chr_lengths_file if it exists.
+    Otherwice look in VCF header for chromosomes.
+    """
     if chr_lengths_file:
         chr_lengths = parse_chr_lengths(chr_lengths_file)
         logger.info("Read length of %d chromosomes from %s", len(chr_lengths), chr_lengths_file)
     else:
         chr_lengths = {
-            chrom: contig.length
+            str(chrom): contig.length
```

Comment:
Did you add this to appease the type checker? I assume that it complained because the keys in `vcf_reader.contigs.items()` can be `None`. If that is the case, it would be better to just filter these out instead of converting them to `"None"`. (That is, add a second condition to the `if contig.length is not None`.)

Reply:
The reason for this was actually that the type for `chrom` was `Union[int, str]` (see https://github.com/pysam-developers/pysam/blob/26f3a7f0fb03cf3347d049f12c8a87ec47acd74e/pysam/libcbcf.pyi#L110). I am not sure why but I agree this maybe not the best solution. After thinking so more I think its better to use the `contig.name` attribute instead of the key since this should be a `str`. Will fix!